### PR TITLE
Another method added for GCD

### PR DESCRIPTION
**Pull request by subahanii**

### CODE REVIEW

File: maths/greatest_common_divisor.py [hunk 1]
```diff
@@ -5,6 +5,15 @@
 """
 
 
+
+# This method is more efficient not acquire more memory cause is no use of any stacks like in recursive as next below mentioned. 
+def gcd_by_iterative(x,y):
+    while y:x,y=y,x%y
```
Comment (cclauss) [inline maths/greatest_common_divisor.py]: Please split this over two lines for readability.
Please add doctests to both functions.  See CONTRIBUTING.md for details.

File: maths/greatest_common_divisor.py [hunk 2]
```diff
@@ -5,21 +5,40 @@
...
         num_2 = int(nums[1])
+        
         print(f"gcd({num_1}, {num_2}) = {gcd(num_1, num_2)}")
+        print(f"By iterative gcd({num_1}, {num_2}) = {gcd_by_iterative(num_1, num_2)}")
+       
     except (IndexError, UnboundLocalError, ValueError):
         print("Wrong Input")
 
-
+
```
Comment (cclauss) [inline maths/greatest_common_divisor.py]: Please lose this whitespace.

**Comment by cclauss** [comment]
https://travis-ci.com/TheAlgorithms/Python/builds/132626798#L384

Also see https://docs.python.org/3/library/doctest.html

**Comment by cclauss** [comment]
https://travis-ci.com/TheAlgorithms/Python/builds/132718197#L371

**Comment by subahanii** [comment]
> https://travis-ci.com/TheAlgorithms/Python/builds/132626798#L384
> 
> Also see https://docs.python.org/3/library/doctest.html

I am not getting where is problem.  Please Marge this all is right.

**Comment by subahanii** [comment]
> https://travis-ci.com/TheAlgorithms/Python/builds/132626798#L384
> 
> Also see https://docs.python.org/3/library/doctest.html

Why you have marked as "Automated test is failing".

**Pull request merged**